File: languages/FSML/Haskell/Language/FSML/QQ/Sample.hs
-- BEGIN ...
{-# LANGUAGE QuasiQuotes #-}
module Language.FSML.QQ.Sample where
import Language.FSML.QuasiQuoter
import Language.FSML.Syntax

-- END ...
sampleFsm :: Fsm
sampleFsm = [fsml|
        initial state locked {
              ticket / collect -> unlocked;
              pass / alarm -> exception;
        }
        state unlocked {
              ticket / eject;
              pass -> locked;
        }
        state exception {
              ticket / eject;
              pass;
              mute;
              release -> locked;
        }
|]

File: languages/FSML/Java/org/softlang/fsml/fluent/Sample.java
// BEGIN ...
package org.softlang.fsml.fluent;

import static org.softlang.fsml.fluent.FsmImpl.fsm;

public class Sample {
	
	public static final
// END ...
		Fsm turnstile =
			fsm()
				.addState("locked")
					.addTransition("ticket", "collect", "unlocked")
					.addTransition("pass", "alarm", "exception")
				.addState("unlocked")
					.addTransition("ticket", "eject", "unlocked")
					.addTransition("pass", null, "locked")
				.addState("exception")
					.addTransition("ticket", "eject", "exception")
					.addTransition("pass", null, "exception")
					.addTransition("mute", null, "exception")
					.addTransition("release", null, "locked");
// BEGIN ...
}
// END ...
